resource/logging_macros.h.em

```diff
@@ -0,0 +1,414 @@
+// generated from rcutils/resource/logging_macros.h.em
+
+// Copyright 2017 Open Source Robotics Foundation, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/*! \file */
+
+#ifndef RCUTILS__LOGGING_MACROS_H_
+#define RCUTILS__LOGGING_MACROS_H_
+
+#include "rcutils/logging.h"
+
+#include <stdio.h>
+#include <stdlib.h>
+
+#if __cplusplus
+extern "C"
+{
+#endif
+
+/**
+ * \def RCUTILS_LOG_MIN_SEVERITY
+ * Define RCUTILS_LOG_MIN_SEVERITY=RCUTILS_LOG_SEVERITY_[DEBUG|INFO|WARN|ERROR|FATAL]
+ * in your build options to compile out anything below that severity.
+ */
+#ifndef RCUTILS_LOG_MIN_SEVERITY
+#define RCUTILS_LOG_MIN_SEVERITY RCUTILS_LOG_SEVERITY_DEBUG
+#endif
+
+// Provide the compiler with branch prediction information
+#ifndef WIN32
+/**
+ * \def RCUTILS_LIKELY
+ * Instruct the compiler to optimize for the case where the argument equals 1.
+ */
+# define RCUTILS_LIKELY(x) __builtin_expect((x), 1)
+/**
+ * \def RCUTILS_UNLIKELY
+ * Instruct the compiler to optimize for the case where the argument equals 0.
+ */
+# define RCUTILS_UNLIKELY(x) __builtin_expect((x), 0)
+#else
+/**
+ * \def RCUTILS_LIKELY
+ * No op since Windows doesn't support providing branch prediction information.
+ */
+# define RCUTILS_LIKELY(x) (x)
+/**
+ * \def RCUTILS_UNLIKELY
+ * No op since Windows doesn't support providing branch prediction information.
+ */
+# define RCUTILS_UNLIKELY(x) (x)
+#endif
+
```

Comment:
I would rather keep it close to the code where it is being used atm. If it turns out that it is useful elsewhere it can be refactored to a common location.